crates/rpc/src/eth_rpc.rs

```diff
@@ -147,7 +148,16 @@ impl EthApiServer for KakarotEthRpc {
     }
 
     async fn transaction_count(&self, _address: Address, _block_number: Option<BlockId>) -> Result<U256> {
-        Ok(U256::from(3))
+        let starknet_block_id = _block_number.map(ethers_block_id_to_starknet_block_id);
+        let starknet_block_id = match starknet_block_id {
```

Comment:
why are we doing this? it feels a bit verbose for no reason

Reply:
We are doing this because the BlockID we get in the arguments of this function are from reth, and we have to convert them to StarknetBlockID types.

We also have to handle the default case of using the latest if the BlockID is `None`.

Is there are some other way you think we can simply this?

Reply:
alright, we can keep it this way for now, as we'll create a wrapper type later!